server/client_manager.py

```diff
@@ -865,6 +865,17 @@ def change_area(self, area, password=""):
                 if not old_area.dark and not old_area.force_sneak:
                     if old_area.area_manager == self.area.area_manager:
                         for c in old_area.clients:
+                            if self.area.area_manager.passing_msg == True:
+                                # Prevents a duplicate message when someone leaves
+                                if count > 0:
+                                    break
+                                if showname == '':
+                                    showname = self.char_name
+                                old_area.send_ic(
+                                    None, '1', 1, "", "", f'~~[º{self.showname}º leaves to º{area.name}º.]', 
+                                    "", "", 1, -1, 0, 0, [0], 0, 0, 0, "", -1, "", "", 0, 0, 0, 0, "0", 0, "", "", "", 0, ""
+                                )
+                                count += 1
```

Comment:
Instead of doing this in the for loop, why not just do it before? You don't have to do the "count" thing that way, simply post all the code before line 867 and it should function